### ISSUE TITLE
it doesn't work on linux

### ISSUE THREAD

**Issue by Vincenzoferrara** [ravibpatel/AutoUpdater.NET#629]
as per the title, i use linux and visual studio code.

i read that it supported net core so i thought it worked.

I'm not using wpf or windows form, but avalonoia ui

error: "resource": "/usr/share/dotnet/sdk/7.0.103/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.DefaultItems.Shared.targets",
"owner": "msCompile",
"code": "NETSDK1136",
"severity": 8,
"message": "Target platform must be set to Windows, typically by including '-windows ' in the TargetFramework property, when using Windows Forms or WPF, or referencing projects or packages that use it.

**Comment by ravibpatel** [comment]
It doesn't support Linux due to its WinForms dependency. I understand the need for Avalonia version. I will try to work on it in my free time to separate update and UI logic, so it can be easier to change UI and use it on another platforms.

**Comment by Vincenzoferrara** [comment]
as i just wanted to add, you could create a version compatible with all systems, and then create another package just for the windows gui

**Comment by ravibpatel** [comment]
Yes, That's the plan. There will be 1 core DLL without any GUI, and other GUI specific libraries just reference it and provide their own GUI. That way, users can create their own GUI easily.
